packages/didcomm/src/modules/credentials/protocol/revocation-notification/services/RevocationNotificationService.ts

```diff
@@ -64,25 +72,61 @@ export class RevocationNotificationService {
       ],
     }
 
-    this.logger.trace('Getting record by query for revocation notification:', query)
-    const credentialRecord = await this.credentialRepository.getSingleByQuery(agentContext, query)
+    this.logger.trace('Getting Credential Exchange record by query for revocation notification:', query)
+    let credentialExchangeRecord = await this.credentialRepository.findSingleByQuery(agentContext, query)
+
+    if (!credentialExchangeRecord) {
+      const w3cCredentialQuery = {
+        $or: [
+          {
+            anonCredsRevocationRegistryId,
+            anonCredsCredentialRevocationId,
+          },
+          {
+            anonCredsUnqualifiedRevocationRegistryId: anonCredsRevocationRegistryId,
+            anonCredsCredentialRevocationId,
+          },
+        ],
+      }
+      this.logger.trace(
+        'Credential Exchange Record not found. Getting W3C credential record by query for revocation notification:',
+        w3cCredentialQuery
+      )
+
+      const w3cCredentialRepository = agentContext.dependencyManager.resolve(W3cCredentialRepository)
+      const w3cCredentialRecord = await w3cCredentialRepository.getSingleByQuery(agentContext, w3cCredentialQuery)
+
+      // Find credential exchange record associated with this credential
+      credentialExchangeRecord =
+        (await this.credentialRepository.getAll(agentContext)).find((record) =>
```

Comment:
Is there no way to find the credential exchange record associated with a credential record?

This seems inefficient to run for every query ... :/

Reply:
Yes, that's why I tried to avoid running it in cases not affected by this problem (e.g. using qualified identifiers).

Probably we have some other tricks to do, but I don't think in practice it will be so critical, since this code runs on holder side and I don't expect users to hold thousands of credentials using unqualified identifiers.

Something not done in this PR (at the moment) are changes on `AnonCredsCredentialFormatService`: we can also add the `anonCredsUnqualified*` tags to CredentialExchangeRecords. This would at least prevent this code from running in 0.5.14 agents.